ci/build.yml

```diff
@@ -6,7 +6,7 @@ image_resource:
   type: docker-image
   source:
     repository: golang
-    tag: 1.19.2
+    tag: latest
```

Comment:
Why latest? Shouldn't docker images be locked to a specific version? Or is that the standard for applications rather than libraries.

Reply:
This is indeed deliberate for libraries. Pinned versions of go are defined in the apps that are using the libraries. As multiple apps may use the same library, it could be used under multiple versions of go. Ideally we'd have libraries tested under multiple versions of go (so run the unit tests under go1.20,1.21,1.22 etc.) but as we can't do that with our current CI implementation we have to choose one. Using the latest version ensures that we've tested the library under the newest go so that once an app is upgraded, we've already tested the library and identified bugs in advance. If we fixed the version here we would run the risk of encountering bugs wile building the app that are really the library's fault.

Reply:
Great answer, thanks.